game/engine.py

```
@@ -403,44 +824,167 @@ def check_game_status(self):
+        The move is validated against the engine before being returned so
+        the AI never plays an illegal book move.
+        """
+        book = self._load_opening_book()
+        fen_key = self.generate_fen_key()
+        candidates = book.get(fen_key)
+        if not candidates:
+            return None
+
+        # Keep the book order stable so the same position always resolves
+        # to the same move across runs.
+        candidates = list(candidates)  # copy – do not mutate the book
```

Ruff (0.15.14)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)